contracts/interchain-token/HyperliquidDeployer.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import { IHyperliquidDeployer } from '../interfaces/IHyperliquidDeployer.sol';
+
+/**
+ * @title HyperLiquidDeployer
+ * @notice This contract explicitly reserves slot 0 for the deployer address.
+ * This is specifically for the linking of ERC 20 tokens between Hyperliquid EVM and Core
+ * where the deployer address used to deploy a spot asset in HyperCore matches the address stored in the first storage slot
+ * of the ERC 20 token deployed in Hyperliquid EVM via create2 mechanism.
+ * Must be inherited first to ensure slot 0 is properly reserved.
+ */
+abstract contract HyperliquidDeployer is IHyperliquidDeployer {
```

Comment:
```suggestion
abstract contract HyperliquidDeployer {
```
use the interface in the token contract instead where both the getter and setter can be implemented